apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/jvm/cpu/CPUProvider.java

```diff
@@ -30,10 +30,11 @@
     CPUProvider() {
         int processorNum = ProcessorUtil.getNumberOfProcessors();
         try {
-            this.cpuMetricsAccessor = (CPUMetricsAccessor) CPUProvider.class.getClassLoader()
-                                                                            .loadClass("org.apache.skywalking.apm.agent.core.jvm.cpu.SunCpuAccessor")
-                                                                            .getConstructor(int.class)
-                                                                            .newInstance(processorNum);
+            //-Dskywalking.cpu_provider=org.apache.skywalking.apm.agent.core.jvm.cpu.JmxCpuAccessor
+            final String cpuAccessorCls = System.getProperty("skywalking.cpu_provider",
```

Comment:
This is a wrong way to extend. 

1. System properties are supported for config injection in the Config.java.
1. Why use try catch and config at the same time?
1. Take a reference from memory gc access. The different accessors should be used based on JVM automatical detection.